### PR TITLE
Request headers

### PR DESCRIPTION
**Pull request by Jude-adesulu**
- Subscription Token authorization changed from params to header

### CODE REVIEW

File: src/subscription/subscription.controller.ts [hunk 1]
```diff
@@ -33,23 +41,39 @@ export class SubscriptionController {
...
   async verify(
-    @Param("token") token: string,
+    @Headers('Authorization') authorization,
     @Body() subscriptionApiCall: SubscriptionApiCallDto,
-  ):Promise<Ok<Object>>{
-    // const verifySubscription = await this.subscriptionService.verifySub(verifysub)
-    const verifySubscription = await this.subscriptionService.makeSubscriptionRequest(token, subscriptionApiCall)
-    return ZapiResponse.Ok(verifySubscription, 'user is subcribed to this api', '200')
+  ): Promise<Ok<Object>> {
+    // const verifySubscreqiption = await this.subscriptionService.verifySub(verifysub)
```
Comment (Chimexx) [inline src/subscription/subscription.controller.ts:51]: Cleanup codes you won't use in future

File: src/subscription/subscription.controller.ts [hunk 2]
```diff
@@ -33,23 +41,39 @@ export class SubscriptionController {
...
+        subscriptionApiCall,
+      );
+    return ZapiResponse.Ok(
+      verifySubscription,
+      'user is subcribed to this api',
+      '200',
+    );
   }
 
   @Get(':profileId/all')
```
Comment (Chimexx) [inline src/subscription/subscription.controller.ts:71]: @Get('all/:profileId')

**Review by Chimexx — APPROVED** [review]

File: src/subscription/subscription.controller.ts (hunk 1 above)
Comment (Jude-adesulu) [reply src/subscription/subscription.controller.ts:51]: That comment wasnt done by me that's why I didn't delete it

File: src/subscription/subscription.controller.ts (hunk 2 above)
Comment (Jude-adesulu) [reply src/subscription/subscription.controller.ts:71]: Same as this, I am not the one that work on it

File: src/subscription/subscription.controller.ts [hunk 3]
```diff
@@ -33,23 +41,39 @@ export class SubscriptionController {
...
     @Body() subscriptionApiCall: SubscriptionApiCallDto,
-  ):Promise<Ok<Object>>{
-    // const verifySubscription = await this.subscriptionService.verifySub(verifysub)
-    const verifySubscription = await this.subscriptionService.makeSubscriptionRequest(token, subscriptionApiCall)
-    return ZapiResponse.Ok(verifySubscription, 'user is subcribed to this api', '200')
+  ): Promise<Ok<Object>> {
+    // const verifySubscreqiption = await this.subscriptionService.verifySub(verifysub)
+    if (!authorization) {
+      throw new BadRequestException(
+        ZapiResponse.BadRequest('No Bearer Token', 'No Token provided', '403'),
```
Comment (SunmadeBabafemi) [inline src/subscription/subscription.controller.ts:54]: I think it was agreed for you not to use the place of the bearer token..... either ways.. it works fine

**Review by SunmadeBabafemi — APPROVED** [review]

File: src/subscription/subscription.controller.ts (hunk 3 above)
Comment (Jude-adesulu) [reply src/subscription/subscription.controller.ts:54]: Yeah speaking of that, I was thinking he never wanted to use Request from express that's why I used the decorator header. When brain sees the code, if he wants a then I would do so cause I don't know what other means to get the token

Comment (PhilemonBrain) [reply src/subscription/subscription.controller.ts:54]: i dont understand u @Jude-adesulu

Comment (Jude-adesulu) [reply src/subscription/subscription.controller.ts:54]: You told me using using request.headers.authorization might cause a problem later may occur later so I used @header instead, I dont know if this idea is okay that's why I said I will wait on your response @PhilemonBrain

Comment (PhilemonBrain) [reply src/subscription/subscription.controller.ts:54]: Isnt that decorator for respoonse?? 
https://docs.nestjs.com/controllers#headers

Comment (PhilemonBrain) [reply src/subscription/subscription.controller.ts:54]: Ohh, sorry my bad. I thought the decorator was above the function. didnt know twas inside.

Comment (PhilemonBrain) [reply src/subscription/subscription.controller.ts:54]: but can't we name it somoething else like 'X-ZAPI-AUTH-TOKEN @Jude-adesulu '

Comment (Jude-adesulu) [reply src/subscription/subscription.controller.ts:54]: You mean I should rename the 'authorization' to X-ZAPI-AUTH-TOKEN?

Comment (Jude-adesulu) [reply src/subscription/subscription.controller.ts:54]: Or the variable name 'token' @PhilemonBrain

**Comment by PhilemonBrain** [comment]
@Jude-adesulu the issue was not just the variable name but the name of the header value in the req.headers
That needs to change as well

**Comment by PhilemonBrain** [comment]
> @Jude-adesulu the issue was not just the variable name but the name of the header value in the req.headers That needs to change as well

Think about it... what difference does changing a 'variable name' make in this code????

**Comment by Jude-adesulu** [comment]
> 

If I'm not mistaken you asked me to change @headers('Authorization') to @headers('X-ZAPI-AUTH-TOKEN'), cause right now I'm confused, have not worked with headers before and I really don't know how the implementation works @PhilemonBrain

**Comment by PhilemonBrain** [comment]
yes, and test it out. @Jude-adesulu 
https://docs.nestjs.com/controllers#request-object

**Comment by Jude-adesulu** [comment]
> yes, and test it out. @Jude-adesulu https://docs.nestjs.com/controllers#request-object

Done sir.

**Comment by PhilemonBrain** [comment]
did u test this? @Jude-adesulu

**Comment by Jude-adesulu** [comment]
> did u test this? @Jude-adesulu
Yes I did. I added the name in my headers plus the token as the value. The way it works now I dont need to split the header since its just the token that will be stored in it without the Bearer variable

**Pull request merged**